packages/flutter/test/material/app_bar_test.dart

```diff
@@ -3416,12 +3416,37 @@ void main() {
         ),
       );
 
-      Color? leadingIconButtonColor() => iconStyle(tester, Icons.menu)?.color;
-      Color? actionIconButtonColor() => iconStyle(tester, Icons.add)?.color;
+      Color? leadingIconButtonColor() => _iconStyle(tester, Icons.menu)?.color;
+      Color? actionIconButtonColor() => _iconStyle(tester, Icons.add)?.color;
 
       expect(leadingIconButtonColor(), Colors.purple);
       expect(actionIconButtonColor(), Colors.purple);
     });
+
+    // This is a regression test for https://github.com/flutter/flutter/issues/130485.
+    testWidgets('Material3 - AppBar.iconTheme correctly applies custom white color in dark mode', (WidgetTester tester) async {
+      final ThemeData themeData = ThemeData(useMaterial3: true, brightness: Brightness.dark);
+      await tester.pumpWidget(
+        MaterialApp(
+          theme: themeData,
+          home: Scaffold(
+            appBar: AppBar(
+              iconTheme: const IconThemeData(color: Colors.white),
+              leading: IconButton(icon: const Icon(Icons.menu), onPressed: () {}),
+              actions: <Widget>[
+                IconButton(icon: const Icon(Icons.add), onPressed: () {}),
+              ],
+            ),
+          ),
+        ),
+      );
+
+      Color? leadingIconButtonColor() => _iconStyle(tester, Icons.menu)?.color;
+      Color? actionIconButtonColor() => _iconStyle(tester, Icons.add)?.color;
+
+      expect(leadingIconButtonColor(), Colors.white);
+      expect(actionIconButtonColor(), Colors.white);
+    });
```

Comment:
Can we also add 3 more unit tests?

1. One similar test for `foregroundColor: Colors.white`.
2. Two tests in the light brightness
```dart
colorScheme: const ColorScheme.light().copyWith(onSurfaceVariant: Colors.red),
```
, the `iconTheme: IconTheme(color: Colors.black87)` and `foregroundColor: Colors.black87` should also be respected. Just want to make sure in light brightness, when we provide the same color(black87) as the `IconTheme.fallback().color` to `IconTheme`, the color can be respected instead of using `onSurfaceVairiant`